app/helpers/application_helper/button/smart_state_scan.rb

```diff
@@ -10,6 +8,14 @@ def check_smart_roles
     end
   end
 
+  def calculate_properties
+    super
+    if disabled?
```

Comment:
Yup, that makes sense :).

We *may* be able to go one step further, *iff* `:enabled => false` is only ever used with `:onwhen => ..`, but .. makes sense as the first step :).